### ISSUE TITLE
Issues with BAM on Blender v 2.76

### ISSUE THREAD

**Issue by Moke2015** [gdbaldw/BlenderAndMBDyn#21]
Dear Doug,

are some scripts broken in the recent version (download yesterday). I can no longer follow your 
tutorials. For example. For the creation of gravity, clicking on the new button triggers a creation
of a 3D drive, which can not be completed, regardless, whether I choose 'Array of 3D drives' or
template drives. A creation of a simple unit drive seems no longer feasible.

Beyond this, because the benefit is plain to see of such an tool, I would happy to contribute if I can.
Where would you require support? 

Best Regards,

Morten

**Comment by Moke2015** [comment]
Hello Doug,

forget my statement above, with regard to the issue. I was probably looking at an obsolete 
tutorial. I did revisit the tutorials and saw that they have been updated. Nevertheless, let me know,
if support is required (and what).

Best Regards,

Morten

**Comment by gdbaldw** [comment]
Welcome Morten,
I'm delighted to know you are making use of this tool. One area where contributions are helpful is in populating all of the MBDyn Elements. For instance, Louis just mentioned a desire for a Total Pin Joint, which is very similar to a Total Joint. May I suggest, if you are familiar with Python and are so moved, copy and paste the Total Joint within element.py, and remake the copy into a Pin Joint.  Let's see how this all develops.
Cheers,
-Doug

**Comment by louisgag** [comment]
Hello Morten,
I'm also very glad to see your proposal for help.
I think Doug's suggestion is great :)
May I ask in which context you use multibody ?
Best Regards,
-Louis

On 21-Oct-2015 11:44 PM, gdbaldw wrote:

> Welcome Morten,
> I'm delighted to know you are making use of this tool. One area where 
> contributions are helpful is in populating all of the MBDyn Elements. 
> For instance, Louis just mentioned a desire for a Total Pin Joint, 
> which is very similar to a Total Joint. May I suggest, if you are 
> familiar with Python and are so moved, copy and paste the Total Joint 
> within element.py, and remake the copy into a Pin Joint. Let's see how 
> this all develops.
> Cheers,
> -Doug
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-150033958.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by Moke2015** [comment]
Hello Doug, Louis,

I have done some fairly simple programming with Python a few years ago. 
I'll have to look
into this and provide feedback.

I will come back to you during the weekend.

Best Regards,

Morten

**Comment by Moke2015** [comment]
Hello Louis,

I have been using MSC.Adams for about 10 years, supporting the design of 
automotive
door latches, which include till today usually DC motor driven 
actuators. I know mbdyn for
some time now. Even though there are some short comings of this software 
for a wide range
of industrial application (collision capabilities, pre- and post 
processing), it is still an
amazing piece of software which I would like to succeed. I am convinced, 
the combination
with Blender is an important step forward.

I hope, I can contribute.

Best Regards,

Morten

**Comment by Moke2015** [comment]
Hello Doug,

I am not familiar with Github. How can I submit a changed code?

Secondly, I had a first look at 'element.py'.  If I am not mistaken, 
class TotalJointOperator(Base) builds the
interface to the GUI, while class TotalJoint(Joint) does the output to 
the mbdyn input file. Is this correct?
Both classes need to be duplicated and referenced, right?

Can you please briefly confirm, whether you recieved this mail? Thanks a 
lot.

Best Regards,

Morten

Am 21.10.2015 um 23:44 schrieb gdbaldw:

**Comment by gdbaldw** [comment]
Hi Moren,
Exactly! Your understanding is perfect, regarding Base and Joint. The one other edit would be in the file, menu.py, to add "Total pin joint" to the menu.
As for github, you would 1) fork BlenderAndMBDyn, and when you are ready 2) submit a pull request. While there are several approaches, what works for me is in Debian to have my git repository at ~/.config/blender/2.72/scripts/addons/BlenderAndMBDyn
I use the following commandline commands often:
git add *.py
git commit -a -m "Description of change"
git push
There's more to it (which I don't recall off hand as I only had to do it once), and github has good help to get you started.
Cheers,
-Doug

**Comment by Moke2015** [comment]
Hello Doug,

sorry to bother you with this inquiry. How did you get the plotting 
running. My installations of blender
(Ubuntu 14.04 + WIn7 Pro) miss obviously the module pandas. The blender 
website did not really
provide good hints in this context.

Best Regards,

Morten

**Comment by gdbaldw** [comment]
Hello Morten,
Install Pandas for Python v2 **and** Pandas for Python v3. Blender, which marshals the Pandas data, uses v3 and Ubuntu, which presents the Pandas data, uses v3. Both v2 and v3 are Debian packages.
Cheers,
-Doug

**Comment by louisgag** [comment]
Dear Morten,
You can follow these instructions to fork Doug's BAMv2: https://help.github.com/articles/fork-a-repo/
Doing something as,
1. create your fork of BAMv2 (from the BAMv2 page)
2. create a local clone of your fork
2a.  sync your local fork to BAMv2: https://help.github.com/articles/syncing-a-fork/
3. makes changes to your clone
4. commit and push to your github
5. create a pull request for Doug to approve.
Best Regards,
-Louis

**Comment by Moke2015** [comment]
Hello Doug,

I had a chance to spend some hours on the code this week (I didn't pull 
yet from Github).
I did copy the entries for the 'Total Joint' to the 'Total Pin Joint'. 
The corresponding entry showed
up in the menu and I could add to the elements list. Nevertheless, when 
I run the simulation -
which creates the mbdyn input file - nothing is saved for the pin joint. 
Do you have any
clue, what is going wrong here?

Beyond this, I have been fighting a few bugs with the Blender interface 
on a three different
computers. While version 2.75 was often crashing on hardware with Radeon 
graphic boards,
2.76 is performing better but still shows some flaws. For example, sub 
menues from BAM
don't show up (Ubuntu 14.04). Running Blender on the same machine with 
WIN7 does not show
this issue. I can provide details / screen dumps later. Overall, the 
robustness of the
blender interface for Linux is somewhat weak.

With regard to Pandas, I did as you wrote. Still no success. The module 
is not found. The same
python version 3.4 is used (bundled and system). I also tried to append 
the python system
path (sys.path.append). Still no luck. Any idea?

Best Regards,

Morten

**Comment by Moke2015** [comment]
Dear Louis, Doug,

I eventually got the Total Pin Joint running (sort of - I still have to 
fix some issues). It took me some time to understand what the code in 
the different files is doing. I should get this done during the next few 
days.
Once this is done, I will fork the GITHUB content, alter it and push it 
back for review.

Best Regards,

Morten

Am 26.10.2015 um 09:11 schrieb louisgag:

**Comment by louisgag** [comment]
Dear Morten,
That's excellent news !
As for pandas, on Ubuntu 14.04 with Blender 2.75a/2.76 I noted the same 
problem as you. I was previously able to plot, but now it doesn't work. 
I'd have to figure out if its due to some BAMv2 update or some Blender 
upgrade.
Regards,
-Louis

On 01-Nov-2015 05:41 PM, Moke2015 wrote:

> Dear Louis, Doug,
> 
> I eventually got the Total Pin Joint running (sort of - I still have to
> fix some issues). It took me some time to understand what the code in
> the different files is doing. I should get this done during the next few
> days.
> Once this is done, I will fork the GITHUB content, alter it and push it
> back for review.
> 
> Best Regards,
> 
> Morten
> 
> Am 26.10.2015 um 09:11 schrieb louisgag:
> 
> > Dear Morten,
> > You can follow these instructions to fork Doug's BAMv2:
> > https://help.github.com/articles/fork-a-repo/
> > Doing something as,
> > 1. create your fork of BAMv2 (from the BAMv2 page)
> > 2. create a local clone of your fork
> > 2a. sync your local fork to BAMv2:
> > https://help.github.com/articles/syncing-a-fork/
> > 3. makes changes to your clone
> > 4. commit and push to your github
> > 5. create a pull request for Doug to approve.
> > Best Regards,
> > -Louis
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > 
> > https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-151055965.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-152842619.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by Moke2015** [comment]
Both,

I would like to understand, how complex the gui has to be.

This simple version is functional now. Nevertheless, the input manual  
allows for additional information (angular orientation in a reference 
frame and globally). So far, I kept it very simpel. What are your thoughts?

Best Regards,

Morten

Am 02.11.2015 um 09:39 schrieb louisgag:

**Comment by louisgag** [comment]
Dear Doug,
I can confirm the bug noted by Morten about the plots no longer working. I've tried on Blender 2.72b, 2.75a, and 2.76.
I get a message of the sort:
 File "/home/louis/.config/blender/2.72/scripts/addons/BlenderAndMBDyn/element.py", line 1150, in invoke
    import pandas as pd
ImportError: No module named 'pandas'
Best Regards,
-Louis

**Comment by louisgag** [comment]
Dear Morten,

I'm assuming your last comment about the complexity of the GUI was in a hope to receive feedback on the TotalPinJoint you are coding. Where can I try it ?  Can you post the code ?

As for my point of view, you could inspire yourself from the level of complexity currently present in the Total Joint.

Best Regards,

-Louis

**Comment by gdbaldw** [comment]
Hi,

Regarding **pandas**, can you confirm that both of these packages are installed on your system?
python-pandas
python3-pandas

If both are installed, and the error occurs, then there is a PATH missing, and perhaps a bug report needs to be submitted against the OS distribution. I'm currently using Debian 8.2.

As for correcting a path issue, this posting provides one approach to isolating and resolving it...
http://stackoverflow.com/questions/7887262/blender3d-python-script-issue-importerror-no-module-named-mymodule

Regarding **sub-menus**, I seem to recall that on occasion hovering over a sub-menu name does not immediately reveal the submenu, but then clicking on the submenu name does reveal the sub-menu.

Regarding **simulation**, simulator.py file line 76 would need to have inserted,
    "Total pin joint",

Cheers,
-Doug

**Comment by louisgag** [comment]
Dear Doug,
I do have both packages installed.
I tried to use bruteforce to resolve the panda issue but it doesn't seem to help. I did both the following mods:

import sys
sys.path.append(r'/usr/lib/python2.7/dist-packages/')

or 

import sys
sys.path.append(r'/usr/lib/python3/dist-packages/')

before

import pandas as pd

but when I try to plot I now get these errors:

  File "/home/louis/.config/blender/2.76/scripts/addons/BlenderAndMBDyn/element.py", line 1153, in invoke
    import pandas as pd
  File "/usr/lib/python3/dist-packages/pandas/**init**.py", line 6, in <module>
    from . import hashtable, tslib, lib
ImportError: cannot import name 'hashtable'

and

  File "/home/louis/.config/blender/2.76/scripts/addons/BlenderAndMBDyn/element.py", line 1153, in invoke
    import pandas as pd
  File "/usr/lib/python2.7/dist-packages/pandas/**init**.py", line 6, in <module>
    from . import hashtable, tslib, lib
ImportError: /usr/lib/python2.7/dist-packages/pandas/hashtable.so: undefined symbol: _Py_ZeroStruct

Best Regards,

-Louis

**Comment by gdbaldw** [comment]
Dear Louis,

I believe a better answer may be,
PYTHONHOME=/usr/lib/python3
in your shell environment before starting blender.

Cheers,
-Doug

**Comment by louisgag** [comment]
Thanks for the tip Doug but the "ImportError: No module named 'pandas'" 
remains.
The strange thing is that a few weeks ago I was plotting without issues.
Regards,
-Louis

On 04-Nov-2015 06:21 PM, gdbaldw wrote:

> Dear Louis,
> 
> I believe a better answer may be,
> PYTHONHOME=/usr/lib/python3
> in your shell environment before starting blender.
> 
> Cheers,
> -Doug
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-153797810.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
Dear Louis,

On further thought, I'm guessing that the issue arises from the fact that Blender uses python3, the shell uses python2.7, and the way I've written plot creates a subprocess in the shell. I believe plot may be rewritten to spawn a daemon thread, which can keep everything in python3. I'll look into this further.

I'm guessing that when the 2.7 / 3 does work, for instance on my system and when it did work on your system, the OS distribution somehow luckily was configured to have Blender find its python3 libraries while the spawned process was able to find its python2.7 libraries.

Cheers,
-Doug

**Comment by gdbaldw** [comment]
Dear Louis,

I believe plotting may now be fixed with the latest update. Plots now use python3 in the subprocess, same as Blender's python version. (As it turns out, a daemon thread is not an option due to a restriction in Tkinter, called by Matplotlib, called by pandas.dataframe.plot()).

Cheers,
-Doug

**Comment by louisgag** [comment]
Dear Doug,

many thanks for your help, but after updating to the latest version I 
still have the same issue. I tried on both 2.72b and 2.76. I also tried 
doing a "export PYTHONHOME=/usr/lib/python3" before running blender. The 
error is still

/Traceback (most recent call last)://
//  File 
"/home/louis/.config/blender/2.76/scripts/addons/BlenderAndMBDyn/element.py", 
line 1144, in invoke//
//    import pandas as pd//
//ImportError: No module named 'pandas'//
//
//location: <unknown location>:-1//
//
//location: <unknown location>:-1/

And as for the pandas packages:

/$ dpkg -l|grep pandas//
//ii  python-pandas 0.13.1-2ubuntu2  
all data structures for "relational" or "labeled" data//
//ii  python-pandas-lib 
0.13.1-2ubuntu2                                     amd64 low-level 
implementations and bindings for pandas//
//ii  python3-pandas 0.13.1-2ubuntu2  
all data structures for "relational" or "labeled" data - Python 3//
//ii  python3-pandas-lib 
0.13.1-2ubuntu2                                     amd64 low-level 
implementations and bindings for pandas - Python 3//
//
//$ dpkg -L python3-pandas//
///.//
///usr//
///usr/share//
///usr/share/doc//
///usr/share/doc/python3-pandas//
///usr/share/doc/python3-pandas/copyright//
///usr/lib//
///usr/lib/python3//
///usr/lib/python3/dist-packages//
///usr/lib/python3/dist-packages/pandas//
///usr/lib/python3/dist-packages/pandas/version.py//
///usr/lib/python3/dist-packages/pandas/tseries//
///usr/lib/python3/dist-packages/pandas/tseries/tests//
//[...]////
///usr/share/doc/python3-pandas/README.md.gz//
///usr/share/doc/python3-pandas/RELEASE.md//
///usr/share/doc/python3-pandas/CONTRIBUTING.md.gz//
///usr/share/doc/python3-pandas/changelog.Debian.gz/

Regards,

-Louis

On 05-Nov-2015 05:17 AM, gdbaldw wrote:

> Dear Louis,
> 
> I believe plotting may now be fixed with the latest update. Plots now 
> use python3 in the subprocess, same as Blender's python version. (As 
> it turns out, a daemon thread is not an option due to a restriction in 
> Tkinter, called by Matplotlib, called by pandas.dataframe.plot()).
> 
> Cheers,
> -Doug
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-153950989.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
From your Linux commandline, does this work...

python3 -c "import pandas"

**Comment by louisgag** [comment]
I'd say it does because it doesn't give me any output.
-L

On 05-Nov-2015 01:58 PM, gdbaldw wrote:

> From your Linux commandline, does this work...
> 
> python3 -c "import pandas"
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-154052013.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
python3 -c "import pandas;print(dir(pandas))"

**Comment by louisgag** [comment]
$ python3 -c "import pandas;print(dir(pandas))"
['Categorical', 'DataFrame', 'DateOffset', 'DateRange', 'DatetimeIndex', 
'ExcelFile', 'ExcelWriter', 'Expr', 'Float64Index', 'HDFStore', 'Index', 
'Int64Index', 'LooseVersion', 'MultiIndex', 'NaT', 'Panel', 'Panel4D', 
'Period', 'PeriodIndex', 'Series', 'SparseArray', 'SparseDataFrame', 
'SparseList', 'SparsePanel', 'SparseSeries', 'SparseTimeSeries', 'Term', 
'TimeGrouper', 'TimeSeries', 'Timestamp', 'WidePanel', '**builtins**', 
'**cached**', '**doc**', '**docformat**', '**file**', '**loader**', 
'**name**', '**package**', '**path**', '**spec**', '**version**', 
'_np_version', '_np_version_under1p6', '_np_version_under1p7', 
'_np_version_under1p8', '_sparse', 'algos', 'bdate_range', 'compat', 
'computation', 'concat', 'core', 'crosstab', 'cut', 'date_range', 
'datetime', 'datetools', 'describe_option', 'eval', 'ewma', 'ewmcorr', 
'ewmcov', 'ewmstd', 'ewmvar', 'ewmvol', 'expanding_apply', 
'expanding_corr', 'expanding_corr_pairwise', 'expanding_count', 
'expanding_cov', 'expanding_kurt', 'expanding_max', 'expanding_mean', 
'expanding_median', 'expanding_min', 'expanding_quantile', 
'expanding_skew', 'expanding_std', 'expanding_sum', 'expanding_var', 
'factorize', 'fama_macbeth', 'get_dummies', 'get_option', 'get_store', 
'groupby', 'hashtable', 'index', 'infer_freq', 'info', 'io', 'isnull', 
'json', 'lib', 'load', 'lreshape', 'match', 'melt', 'merge', 'msgpack', 
'notnull', 'np', 'offsets', 'ols', 'option_context', 'options', 
'ordered_merge', 'pandas', 'parser', 'period_range', 'pivot', 
'pivot_table', 'plot_params', 'pnow', 'qcut', 'read_clipboard', 
'read_csv', 'read_excel', 'read_fwf', 'read_gbq', 'read_hdf', 
'read_html', 'read_json', 'read_msgpack', 'read_pickle', 'read_sql', 
'read_stata', 'read_table', 'reset_option', 'rolling_apply', 
'rolling_corr', 'rolling_corr_pairwise', 'rolling_count', 'rolling_cov', 
'rolling_kurt', 'rolling_max', 'rolling_mean', 'rolling_median', 
'rolling_min', 'rolling_quantile', 'rolling_skew', 'rolling_std', 
'rolling_sum', 'rolling_var', 'rolling_window', 'save', 
'scatter_matrix', 'set_eng_float_format', 'set_option', 'show_versions', 
'sparse', 'stats', 'to_datetime', 'to_msgpack', 'to_pickle', 
'to_timedelta', 'tools', 'tseries', 'tslib', 'unique', 'util', 
'value_counts', 'value_range', 'version', 'wide_to_long']

On 05-Nov-2015 02:14 PM, gdbaldw wrote:

> python3 -c "import pandas;print(dir(pandas))"

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
python3 -c "import subprocess;subprocess.Popen(('python3', '-c', 'import pandas'))"

**Comment by louisgag** [comment]
nothing happens

On 05-Nov-2015 02:21 PM, gdbaldw wrote:

> python3 -c "import subprocess;subprocess.Popen(('python3', '-c', 
> 'import pandas'))"

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
ok.

Next, start blender, switch to the Scripting screen, and at the Blender commandline try import pandas, the same as we had imported BlenderAndMBDyn during out chat. In theory, Popen is supposed to have the same environment as its calling process, and blender is supposed to have the environment from your calling shell.

**Comment by louisgag** [comment]
here it breaks:
"no module name pandas"

On 05-Nov-2015 02:29 PM, gdbaldw wrote:

> ok.
> 
> Next, start blender, switch to the Scripting screen, and at the 
> Blender commandline try import pandas, the same as we had imported 
> BlenderAndMBDyn during out chat. In theory, Popen is supposed to have 
> the same environment as its calling process, and blender is supposed 
> to have the environment from your calling shell.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-154059089.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
Inside blender console
import sys
sys.path

That lists all of your paths for modules. Its supposed to include a path that will reach the pandas module. Mine has many, including /usr/lib/python3/dist-packages/

**Comment by louisgag** [comment]
from that I only get blender "local paths".
It seems to be using a python3.4 lib located here: 
blender-2.76-linux-glibc211-i686/2.76/python/lib/ and provided 
automatically when I downloaded the Blender binaries..!
I've tried creating a symbolic link to my python3 and it didn't help

On 15-11-05 02:38 PM, gdbaldw wrote:

> Inside blender console
> import sys
> sys.path
> 
> That lists all of your paths for modules. Its supposed to include a 
> path that will reach the pandas module.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/gdbaldw/BlenderAndMBDyn/issues/21#issuecomment-154060707.

## 

Louis Gagnon, Ph.D.
Dipartimento di Scienze e Tecnologie Aerospaziali
Politecnico di Milano

**Comment by gdbaldw** [comment]
I found the fix, and posted it as a new issue, solved and closed.

[Solution posted here](https://github.com/gdbaldw/BlenderAndMBDyn/issues/22)

**Comment by jmc8241608** [comment]
Doug,
I recently loaded Blender 2.77 on linux mint. I loaded blenderandmbdyn from github using the BlenderAndMBDy n-master.zip file. I have the mbdyn tab but the options are different from your demos.

How can I fix this?
Jim

It have the following menu items:
Elements
Constitutives
Drivers
Input Cards
Frictions
Matrices
Shapes
Definitions
Simulators

**Comment by gdbaldw** [comment]
Hi,The demos were created during development of the user interface. At this point, the interface is now defined as you see it. I do not immediately recall the differences, but would be glad to answer any specific questions. All the demos can be completed with the new UI, and likely just a few pointers are needed.-Doug  On 04/15/16, jmc8241608notifications@github.com wrote: Doug,I recently loaded Blender 2.77 on linux mint. I loaded blenderandmbdyn from github using the BlenderAndMBDy n-master.zip file. I have the mbdyn tab but the options are different from your demos.How can I fix this?JimIt have the following menu items:ElementsConstitutivesDriversInput CardsFrictionsMatricesShapesDefinitionsSimulators—You are receiving this because you modified the open/close state.Reply to this email directly or view it on GitHub

**Comment by jmc8241608** [comment]
Doug,
Many thanks for the prompt reply! I went ahead and tried the demos on the new interface.  I am struggling with the lack of an Action tab. The simulation tab is set to initial value problem by default. I press the run simulation button and something happens -  but what?  I can't seem to find the the mbdyn  input or output.  How do I access this?
Jim

**Comment by jmc8241608** [comment]
Doug,
The version of Blender 2.77 that I installed is a little unstable.  Could you tell me what the minimum blender version is that I need  to run BlenderAndMBdyn-master.ZIP?
Thanks again!
JIM

**Comment by gdbaldw** [comment]
Jim,Do you have mbdyn installed, from mbdyn.org? The files are located where you save the blender model, in a sub-directory having the same name as your blender model basename.Doug  On 04/15/16, jmc8241608notifications@github.com wrote: Doug,Many thanks for the prompt reply! I went ahead and tried the demos on the new interface.  I am struggling with the lack of an Action tab. The simulation tab is set to initial value problem by default. I press the run simulation button and something happens -  but what?  I can't seem to find the the mbdyn  input or output.  How do I access this?Jim—You are receiving this because you modified the open/close state.Reply to this email directly or view it on GitHub

**Comment by gdbaldw** [comment]
Blender 2.7.2 is the current minimum version. On 04/15/16, jmc8241608notifications@github.com wrote: Doug,The version of Blender 2.77 that I installed is a little unstable.  Could you tell me what the minimum blender version is that I need  to run BlenderAndMBdyn-master.ZIP?Thanks again!JIM—You are receiving this because you modified the open/close state.Reply to this email directly or view it on GitHub

**Comment by jmc8241608** [comment]
Doug.
I regressed my version of blender to 2.72b in order to solve the stability issues.  I did the double pendulum example and got the following error when  I tried to plot the node:

Traceback (most recent call last):
  File "/home/james/.config/blender/2.72/scripts/addons/BlenderAndMBDyn-master/element.py", line 1149, in invoke
    import pandas as pd
ImportError: No module named 'pandas'
Is there a solution?
Jim

**Comment by jmc8241608** [comment]
Doug,
I found the files, thanks. I regressed the version of blender to 2.72b for stability.  I was able to create and run the double pendulum problem.  I received the following error when I tried to plot the node.

 Traceback (most recent call last):
  File "/home/james/.config/blender/2.72/scripts/addons/BlenderAndMBDyn-master/element.py", line 1149, in invoke
    import pandas as pd
ImportError: No module named 'pandas'

Do you know how to fix this?
Thanks,
JIm

-----Original Message-----

**Comment by gdbaldw** [comment]
Jim,As I had mentioned, plotting in Windows requires Pandas be installed and configured to work in the Blender Windows environment. While this should be achievable, I have yet to see anyone provide evidence of a solution or a recipe. This has been discussed on this forum.-Doug On 04/18/16, jmc8241608notifications@github.com wrote: Doug.I regressed my version of blender to 2.72b in order to solve the stability issues.  I did the double pendulum example and got the following error when  I tried to plot the node:Traceback (most recent call last):  File "/home/james/.config/blender/2.72/scripts/addons/BlenderAndMBDyn-master/element.py", line 1149, in invoke    import pandas as pdImportError: No module named 'pandas'Is there a solution?Jim—You are receiving this because you modified the open/close state.Reply to this email directly or view it on GitHub
